dcap/ql/src/quote3.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2022 MobileCoin Foundation
+
+//! Provides functionality for generating quote version 3 quotes
+//!
+//! This functionality requires HW SGX to work correctly otherwise all
+//! functionality will return errors.
+
+use mc_sgx_core_types::Report;
+use mc_sgx_dcap_types::{Quote3, Quote3Error};
+use mc_sgx_util::ResultInto;
+
+/// Create a Quote3 from a Report
+pub trait TryFromReport {
+    /// Try to create a [`Quote3`] from the provided [`Report`]
+    ///
+    /// # Arguments
+    /// * `report` - The report to build the quote from
+    fn try_from_report(report: Report) -> Result<Quote3<Vec<u8>>, Quote3Error> {
```

Comment:
@jcape I know you mentioned [u8] being sufficient for the RawQuote/BaseQuote, wasn't sure if you thought so here too.  If so are you thinking this returns a Vec<u8>.  Due to the sizing issue I'm not fond of the caller having to pass in a pre-sized `[u8]`